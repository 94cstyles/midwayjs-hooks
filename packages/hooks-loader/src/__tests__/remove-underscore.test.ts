import { resolve } from 'path'
import compiler from './compiler'
import webpack from 'webpack'

const root = resolve(__dirname, './fixtures/remove-underscore')

const resolveEntry = (path: string) => {
  return resolve(__dirname, root, 'src/apis/lambda', path)
}
const getOutput = (stats: webpack.Stats) => {
  return stats.toJson().modules[2].source
}

test('Compile normally exported functions', async () => {
  const stats = await compiler(resolveEntry('index.ts'), root)
  const output = getOutput(stats)
  expect(output).toMatchSnapshot()
})

test('Compile normally exported functions', async () => {
  const stats = await compiler(resolveEntry('param.ts'), root)
  const output = getOutput(stats)
  expect(output).toMatchSnapshot()
})
