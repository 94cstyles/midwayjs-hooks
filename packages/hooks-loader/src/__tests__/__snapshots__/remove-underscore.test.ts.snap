// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compile normally exported functions 1`] = `
"import { request } from '@midwayjs/hooks/request'


export  function variableFuncExpression (...args) {
  return request({
    url: '/api/variableFuncExpression',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-variablefuncexpression\\",
  \\"unstable_params\\": [],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function variableFuncExpressionWithArgs (...args) {
  return request({
    url: '/api/variableFuncExpressionWithArgs',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-variablefuncexpressionwithargs\\",
  \\"unstable_params\\": [
    \\"a\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function variableArrowFunction (...args) {
  return request({
    url: '/api/variableArrowFunction',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-variablearrowfunction\\",
  \\"unstable_params\\": [],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function variableArrowFunctionWithArgs (...args) {
  return request({
    url: '/api/variableArrowFunctionWithArgs',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-variablearrowfunctionwithargs\\",
  \\"unstable_params\\": [
    \\"a\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function functionDeclartion (...args) {
  return request({
    url: '/api/functionDeclartion',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-functiondeclartion\\",
  \\"unstable_params\\": [],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function functionDeclartionWithArgs (...args) {
  return request({
    url: '/api/functionDeclartionWithArgs',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-functiondeclartionwithargs\\",
  \\"unstable_params\\": [
    \\"a\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  default  function hasName (...args) {
  return request({
    url: '/api',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index\\",
  \\"unstable_params\\": [
    \\"args\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function dotdotdotArgs (...args) {
  return request({
    url: '/api/dotdotdotArgs',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"index-dotdotdotargs\\",
  \\"unstable_params\\": [
    \\"args\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

"
`;

exports[`Compile normally exported functions 2`] = `
"import { request } from '@midwayjs/hooks/request'


export  function identifier (...args) {
  return request({
    url: '/api/param/identifier',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"param-identifier\\",
  \\"unstable_params\\": [
    \\"a\\",
    \\"b\\",
    \\"c\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function assgin (...args) {
  return request({
    url: '/api/param/assgin',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"param-assgin\\",
  \\"unstable_params\\": [
    \\"a\\",
    \\"b\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function rest (...args) {
  return request({
    url: '/api/param/rest',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"param-rest\\",
  \\"unstable_params\\": [
    \\"a\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

export  function mix (...args) {
  return request({
    url: '/api/param/mix',
    method: 'POST',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"param-mix\\",
  \\"unstable_params\\": [
    \\"a\\",
    \\"b\\",
    \\"c\\"
  ],
  \\"functionGroup\\": \\"fusion-demo\\",
  \\"gateway\\": \\"http\\"
}
  })
}

"
`;
